django/db/backends/oracle/compiler.py

```diff
@@ -17,6 +18,8 @@ def as_sql(self, with_limits=True, with_col_aliases=False):
         do_offset = with_limits and (self.query.high_mark is not None or self.query.low_mark)
         if not do_offset:
             sql, params = super().as_sql(with_limits=False, with_col_aliases=with_col_aliases)
+        elif not self.connection.features.supports_select_for_update_with_limit and self.query.select_for_update:
+            raise NotSupportedError('LIMIT/OFFSET not supported with select_for_update on this database backend.')
```

Comment:
I typically use this style to avoid such long strings near the length limit:
```
raise NotSupportedError(
    '...'
    '...'
)
```
(could also be applied in the other file)